From cd25295064fab517c88d202b5610bd7db34d8812 Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Thu, 27 Apr 2023 19:42:53 +0800
Subject: [PATCH 2/3] nnstreamer: Fix unittest fail

Following unittest use generateTest.py in tests/nnstreamer_filter_pytorch to
generate test data and golden data:

- nnstreamer_converter
- nnstreamer_filter_tensorflow2_lite
- transform_dimchg

But nnstreamer_filter_pytorch might not enable.

So we copy generateTest.py from tests/nnstreamer_filter_pytorch to their unittest install dir.

Signed-off-by: kidd-kw.chen <kidd-kw.chen@mediatek.com>
---
 tests/meson.build                             | 22 +++++++++++++++++++
 tests/nnstreamer_converter/runTest.sh         |  7 +++++-
 .../runTest.sh                                |  7 +++++-
 tests/transform_dimchg/runTest.sh             |  7 +++++-
 4 files changed, 40 insertions(+), 3 deletions(-)

diff --git a/tests/meson.build b/tests/meson.build
index 9218d3eb..662ebdc8 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -343,6 +343,13 @@ if get_option('install-test')
   install_data('gen24bBMP.py', install_dir: unittest_install_dir)
   install_data('test_utils.py', install_dir: unittest_install_dir)
   install_subdir('nnstreamer_converter', install_dir: unittest_install_dir)
+
+  # Workaround:
+  # nnstreamer_converter unittest will calls generateTest.py in nnstreamer_filter_pytorch to generate testing data but pytorch might be disable.
+  # So we copy generateTest.py from nnstreamer_filter_pytorch to nnstreamer_converter and also make unittest to use this generateTest.py.
+  dir = join_paths(unittest_install_dir, 'nnstreamer_converter')
+  install_data(sources : 'nnstreamer_filter_pytorch/generateTest.py', install_dir : dir)
+
   install_subdir('nnstreamer_merge', install_dir: unittest_install_dir)
   install_subdir('nnstreamer_decoder', install_dir: unittest_install_dir)
   install_subdir('nnstreamer_decoder_boundingbox', install_dir: unittest_install_dir)
@@ -357,7 +364,15 @@ if get_option('install-test')
   endif
   if tflite2_support_is_available
     install_subdir('nnstreamer_filter_tensorflow2_lite', install_dir: unittest_install_dir)
+
+    # Workaround:
+    # tflite2 unittest will calls generateTest.py in nnstreamer_filter_pytorch to generate testing data but pytorch might be disable.
+    # So we copy generateTest.py from nnstreamer_filter_pytorch to nnstreamer_filter_tensorflow2_lite and also make unittest to use this generateTest.py.
+    dir = join_paths(unittest_install_dir, 'nnstreamer_filter_tensorflow2_lite')
+    install_data(sources : 'nnstreamer_filter_pytorch/generateTest.py', install_dir : dir)
   endif
+
+
   if have_python3
     if flatbuf_support_is_available
       install_subdir('nnstreamer_converter_python3', install_dir: unittest_install_dir)
@@ -397,6 +412,13 @@ if get_option('install-test')
   install_subdir('transform_arithmetic', install_dir: unittest_install_dir)
   install_subdir('transform_clamp', install_dir: unittest_install_dir)
   install_subdir('transform_dimchg', install_dir: unittest_install_dir)
+
+  # Workaround:
+  # transform_dimchg unittest will calls generateTest.py in nnstreamer_filter_pytorch to generate testing data but pytorch might be disable.
+  # So we copy generateTest.py from nnstreamer_filter_pytorch to transform_dimchg and also make unittest to use this generateTest.py.
+  dir = join_paths(unittest_install_dir, 'transform_dimchg')
+  install_data(sources : 'nnstreamer_filter_pytorch/generateTest.py', install_dir : dir)
+
   install_subdir('transform_stand', install_dir: unittest_install_dir)
   install_subdir('transform_transpose', install_dir: unittest_install_dir)
   install_subdir('transform_typecast', install_dir: unittest_install_dir)
diff --git a/tests/nnstreamer_converter/runTest.sh b/tests/nnstreamer_converter/runTest.sh
index ff928789..3bc944a2 100644
--- a/tests/nnstreamer_converter/runTest.sh
+++ b/tests/nnstreamer_converter/runTest.sh
@@ -24,7 +24,12 @@ if [ "$SKIPGEN" == "YES" ]; then
 else
     echo "Test Case Generation Started"
     python3 generateGoldenTestResult.py
-    python3 ../nnstreamer_filter_pytorch/generateTest.py
+    
+    # Workaround:
+    # unittest will calls generateTest.py in nnstreamer_filter_pytorch to generate testing data but pytorch might be disable.
+    # So we copy generateTest.py from nnstreamer_filter_pytorch to here and also make unittest to use this generateTest.py.
+    #python3 ../nnstreamer_filter_pytorch/generateTest.py
+    python3 generateTest.py
     sopath=$1
 fi
 convertBMP2PNG
diff --git a/tests/nnstreamer_filter_tensorflow2_lite/runTest.sh b/tests/nnstreamer_filter_tensorflow2_lite/runTest.sh
index f28d4b0c..7e44c556 100644
--- a/tests/nnstreamer_filter_tensorflow2_lite/runTest.sh
+++ b/tests/nnstreamer_filter_tensorflow2_lite/runTest.sh
@@ -70,7 +70,12 @@ if [ "$SKIPGEN" == "YES" ]; then
     sopath=$2
 else
     echo "Test Case Generation Started"
-    python3 ../nnstreamer_filter_pytorch/generateTest.py
+
+    # Workaround:
+    # unittest will calls generateTest.py in nnstreamer_filter_pytorch to generate testing data but pytorch might be disable.
+    # So we copy generateTest.py from nnstreamer_filter_pytorch to here and also make unittest to use this generateTest.py.
+    #python3 ../nnstreamer_filter_pytorch/generateTest.py
+    python3 generateTest.py
     sopath=$1
 fi
 
diff --git a/tests/transform_dimchg/runTest.sh b/tests/transform_dimchg/runTest.sh
index 0d2d9ae7..3f511c03 100644
--- a/tests/transform_dimchg/runTest.sh
+++ b/tests/transform_dimchg/runTest.sh
@@ -26,7 +26,12 @@ if [ "$SKIPGEN" == "YES" ]; then
 else
     echo "Test Case Generation Started"
     python3 ../nnstreamer_converter/generateGoldenTestResult.py 8
-    python3 ../nnstreamer_filter_pytorch/generateTest.py
+    
+    # Workaround:
+    # unittest will calls generateTest.py in nnstreamer_filter_pytorch to generate testing data but pytorch might be disable.
+    # So we copy generateTest.py from nnstreamer_filter_pytorch to here and also make unittest to use this generateTest.py.
+    #python3 ../nnstreamer_filter_pytorch/generateTest.py
+    python3 generateTest.py
     sopath=$1
 fi
 convertBMP2PNG
-- 
2.18.0

